MobileSdk/src/main/java/com/spruceid/mobile/sdk/oid4vp/CredentialCallback.kt

```diff
@@ -0,0 +1,53 @@
+package com.spruceid.mobile.sdk.oid4vp
+
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.material3.Button
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import com.spruceid.mobile.sdk.rs.CredentialCallback
+import com.spruceid.mobile.sdk.rs.ParsedCredential
+import com.spruceid.mobile.sdk.rs.PermissionRequest
+import com.spruceid.mobile.sdk.rs.PermissionResponse
+import com.spruceid.mobile.sdk.rs.Uuid
+
+@Composable
+fun SelectCredentialsView(
+    credentials: List<ParsedCredential>,
+    onSelectedCredential: (ParsedCredential) -> Unit
+) {
+    return LazyColumn {
+        items(credentials.size) { credential ->
+            val cred = credentials[credential]
+
+            CredentialItem(cred, onClick = { onSelectedCredential(cred) })
+        }
+    }
+}
+
+@Composable
+fun CredentialItem(credential: ParsedCredential, onClick: () -> Unit) {
+    Button(onClick = onClick) { Text(text = credential.format().toString()) }
+}
+
+class CredentialPresentation(
+//    onCredentials handler to called when the Rust code calls the permissionRequest method
+    private val onCredentials: (List<ParsedCredential>) -> Unit,
+//    callback to determine if the credential has been selected.
+    private var selectedCredential: () -> Uuid?
+): CredentialCallback {
+
+    private var credentials: List<ParsedCredential> = emptyList()
+
+    override suspend fun permissionRequest(request: PermissionRequest) {
+        onCredentials(request.credentials())
+    }
+
+    // This method will be called when the Rust code awaits the response.
```

Comment:
The while statement likely will block the current thread. Split the `PermissionResponse` handler as a separate method.